dev/cmake/cmake/FindCriterion.cmake

```diff
@@ -1,27 +1,100 @@
+#   Criterion_LIBRARIES - Libraries needed to link to Criterion
+#   Criterion_VERSION - The version of the Criterion which was found
+#
+# Cache variables:
+#   Criterion_INCLUDE_DIR - The directory containing criterion/criterion.h
+#   Criterion_LIBRARY - The path to the Criterion library
+#
+# Example usage:
+#   find_package(Criterion)
+#   target_link_libraries(foo PRIVATE Criterion::Criterion)
+
+set(_criterionReason "")
+
+# Try pkg-config.
+find_package(PkgConfig QUIET)
+if(PKG_CONFIG_FOUND)
+  pkg_check_modules(PC_Criterion QUIET criterion)
+endif()
 
-find_package(PkgConfig)
+find_path(
+  Criterion_INCLUDE_DIR
+  criterion/criterion.h
+  HINTS ${PC_Criterion_INCLUDE_DIRS}
+  DOC "The directory containing criterion/criterion.h"
+)
+mark_as_advanced(Criterion_INCLUDE_DIR)
 
-find_path(CRITERION_INCLUDE_DIR criterion/criterion.h
-          PATH_SUFFIXES criterion)
+if(NOT Criterion_INCLUDE_DIR)
+  string(APPEND _criterionReason "criterion/criterion.h not found. ")
+endif()
 
-find_library(CRITERION_LIBRARY NAMES criterion libcriterion)
+find_library(
+  Criterion_LIBRARY
+  NAMES criterion libcriterion
+  HINTS ${PC_Criterion_LIBRARY_DIRS}
+  DOC "The path to the Criterion library"
+)
+mark_as_advanced(Criterion_LIBRARY)
 
-set(CRITERION_LIBRARIES ${CRITERION_LIBRARY})
-set(CRITERION_INCLUDE_DIRS ${CRITERION_INCLUDE_DIR})
+if(NOT Criterion_LIBRARY)
+  string(APPEND _criterionReason "Criterion library (libcriterion) not found. ")
+endif()
+
+# Get Criterion version from pkg-config if available. Conditional check ensures
+# that Criterion found by pkg-config is the one found by the find_path() and
+# find_library().
+if(PC_Criterion_VERSION AND Criterion_INCLUDE_DIR IN_LIST PC_Criterion_INCLUDE_DIRS)
+  set(Criterion_VERSION ${PC_Criterion_VERSION})
+endif()
 
 include(FindPackageHandleStandardArgs)
-# handle the QUIET and REQUIRED arguments and set CRITERION_FOUND to TRUE
-# if all listed variables are TRUE
-find_package_handle_standard_args(Criterion DEFAULT_MSG
-                                  CRITERION_LIBRARY CRITERION_INCLUDE_DIR)
+# Handle the REQUIRED, QUIET, and version-related arguments of find_package().
+find_package_handle_standard_args(
+  Criterion
+  REQUIRED_VARS Criterion_LIBRARY Criterion_INCLUDE_DIR
+  VERSION_VAR Criterion_VERSION
+  HANDLE_VERSION_RANGE
+  REASON_FAILURE_MESSAGE "${_criterionReason}"
+)
+
+unset(_criterionReason)
+
+if(NOT Criterion_FOUND)
+  return()
+endif()
+
+set(Criterion_INCLUDE_DIRS ${Criterion_INCLUDE_DIR})
+set(Criterion_LIBRARIES ${Criterion_LIBRARY})
+
+if(NOT TARGET Criterion::Criterion)
+  if(IS_ABSOLUTE "${Criterion_LIBRARY}")
+    add_library(Criterion::Criterion UNKNOWN IMPORTED)
+    set_target_properties(
+      Criterion::Criterion
+      PROPERTIES IMPORTED_LOCATION "${Criterion_LIBRARY}"
+    )
+  else()
+    add_library(Criterion::Criterion INTERFACE IMPORTED)
+    set_target_properties(
+      Criterion::Criterion
+      PROPERTIES IMPORTED_LIBNAME "${Criterion_LIBRARY}"
```

Comment:
I think `INTERFACE_INCLUDE_DIRECTORIES` should be set too in both cases.

Reply:
Hello, this one is also set at the end of the module in this file.

Reply:
Indeed. Sorry for the oversight.